Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp

```diff
@@ -98,24 +91,23 @@ void NetworkMDNSRegister::closeAndForgetService(DNSServiceRef service)
     });
 }
 
-static void registerMDNSNameCallback(DNSServiceRef service, DNSRecordRef record, DNSServiceFlags, DNSServiceErrorType errorCode, void *context)
+static void registerMDNSNameCallback(DNSServiceRef service, DNSRecordRef record, DNSServiceFlags, DNSServiceErrorType errorCode, void* context)
 {
-    auto request = pendingRegistrationRequests().take(reinterpret_cast<uintptr_t>(context));
-    if (!request)
-        return;
+    auto* request = reinterpret_cast<PendingRegistrationRequest*>(context);
 
     MDNS_RELEASE_LOG_IN_CALLBACK(request->sessionID, "registerMDNSNameCallback with error %d", errorCode);
 
     if (errorCode) {
         request->connection->mdnsRegister().closeAndForgetService(service);
-        request->connection->connection().send(Messages::WebMDNSRegister::FinishedRegisteringMDNSName { request->requestIdentifier, request->name, WebCore::MDNSRegisterError::DNSSD }, 0);
-
+        request->completionHandler(request->requestIdentifier, request->name, WebCore::MDNSRegisterError::DNSSD);
+        delete request;
```

Comment:
Why can't we have a simple std::unique_ptr local inside this function instead of a ScopeExit? I'd prefer a std::unique_ptr unless there is a good reason not to.